p2p/net/conn/listen.go

```diff
@@ -144,9 +145,9 @@ func Listen(ctx context.Context, addr ma.Multiaddr, local peer.ID, sk ic.PrivKey
 		Listener: ml,
 		local:    local,
 		privk:    sk,
-		cg:       ctxgroup.WithContext(ctx),
+		proc:     goprocessctx.WithContext(ctx),
 	}
-	l.cg.SetTeardown(l.teardown)
+	l.proc.SetTeardown(l.teardown)
```

Comment:
WithContextAndTeardown is still not necessary.
The fix here is to SetTeardown right after proc initiation, making it like a WithContextAndTeardown.

But this pattern happens at least twice now, if this is considered frequent enough then I will make one.